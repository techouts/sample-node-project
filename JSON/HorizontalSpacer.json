{
  "id": 35,
  "__component": "widget.horizontal-spacer",
  "bgColor": null,
  "height": "40px"
}