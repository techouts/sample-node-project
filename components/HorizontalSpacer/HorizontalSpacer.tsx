import { Box } from "@mui/material";
import React from "react";
import HorizontalSpacerSchema from "../../schemas/HorizontalSpacerSchema";

function HorizontalSpacer({
  __component,
  id,
  bgColor,
  height,
}: HorizontalSpacerSchema) {
  return (
    <Box
      height={`${height}`}
      width={"100%"}
      sx={{ backgroundColor: bgColor ? bgColor : "transparent" }}
      className={`${__component}_${id}`}
    />
  );
}

export default HorizontalSpacer;
