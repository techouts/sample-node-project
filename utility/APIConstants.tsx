const NEXT_PUBLIC_JUSPAY_APIURL = process.env.NEXT_PUBLIC_JUSPAY_APIURL;
export const merchant_id = "ssbeauty";
export const oauth_token = "authorizationserver/oauth/token";
export const otp_services = "services/v2_1/ssl/sendOTP/OB";
export const otp_service_verifier =
  "services/v2_1/ssl/users/loginOrSignup?mobileNumber=9640558040&otp=782253&cliend_id=mobile_android";
export const SignUp_client_id = "mobile_android";
export const getPaymentMethods = `${NEXT_PUBLIC_JUSPAY_APIURL}/merchants/${merchant_id}/paymentmethods?options.add_emandate_payment_methods=false&options.check_wallet_direct_debit_support=false&options.add_supported_reference_ids=false`;
export const CardBin = `${NEXT_PUBLIC_JUSPAY_APIURL}/cardbins/`;
export const CardTokenization = `${NEXT_PUBLIC_JUSPAY_APIURL}/card/tokenize`;
export const CardTransaction = `${NEXT_PUBLIC_JUSPAY_APIURL}/txns`;
export const QyPaymentStatus = `${NEXT_PUBLIC_JUSPAY_APIURL}/order/payment-status?merchant_id=ssbeauty&order_id=`;
export const paymentStatus =
  "/order/payment-status?merchant_id=ssbeauty&order_id=";
export const GOOGLE_MAP_DIRECTIONS_API = (
  origin: string,
  destination: string
) =>
  `${process.env.NEXT_PUBLIC_GOOGLE_MAP_URL}?api=1&origin=${origin}&destination=${destination}&travelmode=car`;
  export const UNBXD_HOST = process.env.NEXT_UNBXD_HOST;

  export const IS_UNBXD_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_ENABLED === "true";

  export const MIDDLEWARE_UNBXD = process.env.NEXT_PUBLIC_MIDDLEWARE_UNBXD_SEARCH;

  export const IS_UNBXD_MIDDLEWARE_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_MIDDLEWARE_ENABLED === "true"

  const NEXT_PUBLIC_D365_APP_ID_GUEST = process.env.NEXT_PUBLIC_D365_APP_ID_GUEST;

  const NEXT_PUBLIC_D365_APP_ID_LOGGED_IN = process.env.NEXT_PUBLIC_D365_APP_ID_LOGGED_IN;

  const NEXT_PUBLIC_D365_ORG_ID = process.env.NEXT_PUBLIC_D365_ORG_ID;
  
  const NEXT_PUBLIC_D365_ORG_URL = process.env.NEXT_PUBLIC_D365_ORG_URL;
  export const Google_API_KEY = "AIzaSyCBo5PdGTw5Z0BS6hTboacbxkMC7vIZ6FY";

  export {NEXT_PUBLIC_D365_APP_ID_LOGGED_IN, NEXT_PUBLIC_D365_APP_ID_GUEST, NEXT_PUBLIC_D365_ORG_ID, NEXT_PUBLIC_D365_ORG_URL}
  export const IS_RECAPTCHA_ENABLED = process.env.NEXT_PUBLIC_IS_RECAPTCHA_ENABLED === "true";