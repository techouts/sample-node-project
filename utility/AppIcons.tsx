export const Camera_ICON = "Camera_icon";
export const EDIT_IMAGE_ICON = "EDIT_IMAGE";
export const InfoCircle_Grey_ICON = "InfoCircle_Grey";
export const Profile_Add_ICON = "Profile_Add";

export const TIER0_ICON = "TIER0_LOGO";
export const TIER1_ICON = "TIER1_LOGO";
export const TIER2_ICON = "TIER2_LOGO";
export const TIER3_ICON = "TIER3_LOGO";
export const TIER4_ICON = "TIER4_LOGO";

export const GET_APP_LOGO_HEADER = "Get_App";
export const GIFT_CARD_ICON = "Gift_Card";
export const STORES_AND_EVENTS_ICON = "Stores_And_Events";
export const HELP_ICON = "Help";
export const USER_SIGNIN_ICON = "user_signin";
export const LOCATION_ICON = "Location";
export const DOWN_ARROW_ICON = "ARROWDOWN";
export const DELETE_ICON = "Delete_icon";
export const ADD_ADDRESS_ICON = "Add_address";
export const CHECKED_ICONS = "CHECKED_ICON";
export const UNCHECKED_ICONS = "UNCHECKED_ICON";
export const BeautyProfilecheck_ICON = "BeautyProfilecheck";
export const EDIT_PINK_ICON = "EditIcon_PinkColour";
export const CHECKED_RADIO_ICON = "Selected_Radio_Button";
export const UNCHECKED_RADIO_ICON = "Unselected_Radio_Button";
export const VERIFIED_IMAGE_ICON = "VERIFIED_IMAGE";
export const TICK_CIRCLE_WARNING = "TickCircleWarning";
export const COD_ICON = "CODImage";
export const CART_TRUCK_ICON = "CartTruckImage";
export const EMPTY_WISHLIST_COLOUR = "Wishlist_Empty_Colour";
export const NORMAL_SEARCH_ICON = "SEARCH_NORMAL";
export const FILTER_SETTINGS_ICON = "Filter_settings";
export const REFRESH_ICONS = "Refresh_icon";
export const SPEAKER_ICONS = "Speaker_icon";
export const CARD_PAYMENT_IMAGE = "CardPaymentImage";
export const AIRTEL_MOBILE_IMAGE = "Airtel_image_mobile";
export const MOBIWICK_MOBILE_IMAGE = "Mobiwick_mobile_image";
export const FREECHARGE_MOBILE_IMAGE = "Freecharge_mobile_image";
export const JIO_MOBILE_IMAGE = "Jio_mobile_image";
export const ZOOM_ICON = "ZoomIconUrl";
export const SHARE_ICONS = "Share_Icon";
export const WISHLIST_ADDED_ICON = "Wishlist_filled_heart";
export const PLAY_VIDEO_ICON = "VideoPlayButton";
export const LEFT_ARROW_ICON = "Left_direction_icon";
export const USER_ACCOUNT_ICON = "account_user_icon"
export const CLOSE_CIRCLE_ICON = "close-circle";
export const LIKE_ICONS = "LIKE_ICON";
export const UNLIKE_ICONS = "UNLIKE_ICON";
export const FILLED_TICK_CIRLE_ICON = "Filled_tick_circle";
export const Closecircle_BlackColour_ICON = "Closecircle_BlackColour";
export const current_location_pointer_ICON = "current_location_pointer";
export const Empty_wishlist_webicon_icon = "Empty_wishlist_webicon";
export const Empty_wishlist_appbgicon_icon = "Empty_wishlist_appbgicon";
export const CLOSE_CIRCLE_IMAGE = "close_circle";
export const DISCOUNTS_ICON = "discount_shape";
export const SORT_ICON = "SORT_ARROW";
export const EMPTY_STAR_ICON = "EmptyStar";
export const TROPHY_ICONS = "Trophy_icon";
export const GPS_LOCATION_ICON = "Gps_Pink_Icon";
export const HEADER_SEARCH_ICON = "headerSearchIcon";
export const HEADER_WISHLIST_ICON = "headerWishlistIcon";
export const HEADER_CART_ICON = "headerCartIcon";
export const BRAND_UP_ARROW = "BRAND_UP_ARROW";
export const BRAND_DOWN_ARROW = "BRAND_DOWN_ARROW";
export const HEADER_BACK_ARROW = "headerNavigationIcon";
export const INFO_ICON = "INFO_ICON";
export const FALL_BACK_IMAGE_PRODUCT = "PRODUCT_FALLBACK_URL";
export const Error_Image = "ErrorImage";
export const No_RESULTS_SEARCH_ICON = "noResultsSearchIcon";
export const RECENT_SEARCH_ICON =
  "https://storage.googleapis.com/images_cms_sit_ssb/recent_clock_bde48d7c4e_1_986324a98d.svg";