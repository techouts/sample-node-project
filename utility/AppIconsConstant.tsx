import { useRecoilState } from "recoil";
import { SSBLogos } from "../recoilstore";

export const AppIcons = (image: string) => {
  const [SSBeautyLogos] = useRecoilState(SSBLogos);

  const imageIcon = SSBeautyLogos?.appLogos?.find(
    (item: any) => item?.name == image
  );
  return imageIcon;
};
