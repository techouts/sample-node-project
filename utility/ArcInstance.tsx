import axios from "axios";
import { Cookies } from "react-cookie";

const cookie = new Cookies();
const NEXT_PUBLIC_ORC_LAYER = process.env.NEXT_PUBLIC_ORC_LAYER;

export const arcClient = axios.create({
  baseURL: NEXT_PUBLIC_ORC_LAYER,
});

arcClient.interceptors.request.use((config) => {
  const accessToken = `${cookie.get("accessToken")}`;
  const axiosConfig = { ...config };
  axiosConfig.headers["Content-Type"] = "application/json";
  axiosConfig.headers["Access-Control-Allow-Origin"] = "*";
  axiosConfig.headers["Accept"] = "*/*";
  axiosConfig.headers["Accept-Encoding"] = "gzip, deflate, br";
  axiosConfig.headers["Connection"] = "keep-alive";
  if (accessToken && accessToken !== "undefined") {
    axiosConfig.headers.Authorization = `Bearer ${accessToken}`;
  }
  return axiosConfig;
});