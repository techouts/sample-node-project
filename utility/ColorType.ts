import { type } from "os";

type RGB = `rgb(${number}, ${number}, ${number})`;
type RGBA = `rgba(${number}, ${number}, ${number}, ${number})`;
type HEX = `#${string}`;

export type ColorType = RGB | RGBA | HEX;