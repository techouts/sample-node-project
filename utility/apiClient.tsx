import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
import { callErrorEvent } from "./ErrorEventAnalytics";
import { fetchMiddleWareAccessToken } from "./MiddelwareAccessWrapper";

const GRAPHQL_URL = process.env.NEXT_PUBLIC_GRAPHQL_URL!;
const MIDDLEWARE_URL = process.env.NEXT_PUBLIC_SSO_API_GATEWAY_URL!;
const isMiddlewareSecurityEnabled =
  process.env.NEXT_PUBLIC_IS_MIDDLEWARE_SECURITY_ENABLED === "true";

const axiosInstance = axios.create();

async (config: AxiosRequestConfig<any>) => {
  if (!config.url?.includes("/middleware")) return config;

  const token = localStorage.getItem("accessToken");
  const updatedHeaders: AxiosRequestConfig["headers"] = {
    ...config.headers,
    authorization: token ? `Bearer ${token}` : "",
  };
  if (isMiddlewareSecurityEnabled) {
    let mToken = localStorage.getItem("m-token");
    if (mToken && mToken !== "") {
      updatedHeaders["x-auth-token"] = `Bearer ${mToken}`;
    } else {
      const mTokenRes = await fetchMiddleWareAccessToken();
      if (mTokenRes?.data?.access_token) {
        updatedHeaders[
          "x-auth-token"
        ] = `Bearer ${mTokenRes?.data?.access_token}`;
      }
    }
  }

  return { ...config, headers: updatedHeaders };
};

axiosInstance.interceptors.response.use(
  (response: AxiosResponse) => {
    return response;
  },
  async (error: any) => {
    if (error.response) {
      const { message, status } = error.response.data || {};
      console.error(`[Error]: ${message}, Status: ${status}`);

      let params = {
        error_message: message || error.message,
        status: status || error.response?.status,
        item_name: GRAPHQL_URL,
      };

      callErrorEvent(params);
    } else if (error.request) {
      console.error("[Network error]: No response received");
    } else {
      console.error(`[Error]: ${error.message}`);
    }
    return Promise.reject(error);
  }
);

// Retry logic for requests
const retryRequest = async (
  config: AxiosRequestConfig<any>,
  retries = 3
): Promise<AxiosResponse | void> => {
  let lastError;

  for (let i = 0; i < retries; i++) {
    try {
      return await axiosInstance(config);
    } catch (error: any) {
      lastError = error;

      if (error.response?.status === 401 || error.response?.status === 403) {
        console.warn(
          `Attempt ${i + 1}: Received ${
            error.response.status
          }, trying to refresh token...`
        );
        const mTokenRes = await fetchMiddleWareAccessToken();
        const mToken = mTokenRes?.data?.access_token;

        if (!mToken) {
          console.error("Token refresh failed. Aborting retries.");
          break;
        }
        localStorage.setItem("m-token", mToken);
        config = {
          ...config,
          headers: {
            ...(config.headers || {}),
            "x-auth-token": `Bearer ${mToken}`,
          },
        };

        continue;
      }

      console.error(
        `Request failed with status ${error.response?.status}. Retrying...`
      );
    }
  }

  return Promise.reject(lastError);
};

// Function to send requests
const sendRequest = async (
  query: string,
  isCMSQuery = false,
  middleWareURL = false,
  urlKey?: string
): Promise<AxiosResponse | void> => {
  const url = urlKey
    ? `${urlKey}?query=${query}`
    : isCMSQuery
    ? query
    : middleWareURL
    ? `${MIDDLEWARE_URL}?query=${query}`
    : `${GRAPHQL_URL}?query=${query}`;

  const config: AxiosRequestConfig = {
    method: "get",
    url: url,
    headers: {},
  };

  try {
    return await retryRequest(config);
  } catch (error: any) {
    console.error(`Request failed: ${error}`);
    return Promise.reject(error);
  }
};

export default sendRequest;
