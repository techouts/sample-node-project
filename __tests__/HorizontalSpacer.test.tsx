import { render } from "@testing-library/react";
import HorizontalSpacer from "../components/HorizontalSpacer/HorizontalSpacer";

import { ColorType } from "../utility/ColorType";
const color: ColorType = "#0000";

const horizontalSpacerData = {
  id: 10,
  __component: "widget.horizontal-spacer",
  bgColor: color,
  height: "15px",
  display: "web",
  position: 24,
};

describe("Horizontal Spacer", () => {
  it("renders a horizontal spacer component", () => {
    const { container } = render(
      <HorizontalSpacer {...horizontalSpacerData} />
    );
    expect(container).toMatchSnapshot();
  });
});
