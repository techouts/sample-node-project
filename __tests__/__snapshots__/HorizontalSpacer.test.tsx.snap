// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horizontal Spacer renders a horizontal spacer component 1`] = `
<div>
  <div
    class="widget.horizontal-spacer_10 MuiBox-root css-4evupm"
  />
</div>
`;
